packages/flutter_tools/lib/src/template.dart

```diff
@@ -26,13 +27,35 @@ import 'dart/package_map.dart';
 /// but without template expansion (data files, etc.), the '.copy.tmpl'
 /// extension may be used.
 ///
+/// Furthermore, templates may contain test files and golden images that are only
+/// included when the --include-tests flag is enabled. Test files must end in
+/// `.test.tmpl`. Golden images must end in '.golden.tmpl'. Golden images are
+/// copied from the flutter_template_images repository.
```

Comment:
the images themselves can't be landed in this repo. cc @Piinks who may be able to show you how to use skia gold to do this.

Reply:
Are the golden file tests being run here in our CI? Or are these tests supposed to be run by users locally after creating their own project?

Reply:
Hi @Piinks :) 

Primary Goal: The Golden Files were added to verify this new template functions correctly on the CI before shipping to users. 

Secondary Goal: During the development of the template, some users (as well as @zanderso above!) asked for more comprehensive test coverage in the template as an educational tool. Therefore, allowing users to generate a template with some Golden files was only meant to be a "bonus" for those curious.

I've taken a look at Flutter Gold, but it appears it is purpose-built for `package:flutter`, whereas this PR modifies `package:flutter_tools`. Is there any documentation on how to adapt that process for Flutter tools?

Reply:
I see! Just to clarify, is the expectation that the same golden file tests will run one way here on our CI and another way when the user creates their own app locally? You are right, flutter tools is not set up to use Gold right now, but I think it could be done. The test config would need to set up a special file comparator that will talk to gold for CI, but that would not work locally.

I am wondering, if the same tests are meant to execute differently this way, and we have a special config in the template to run on our CI in flutter/flutter, will that be confusing to users that see that as an example and think it is how they should set up their tests? 

Or, would it be possible to create a test in the flutter package that creates this app and keeps a golden file of it, and then the template itself can just ship a set of golden file tests that work as the user would expect them? Instead of including golden files here, you could have `flutter test --update-goldens` run when the template is created and it will generate the golden files for users to see and test against. That way we have tests covering this for verifying the template, but users looking to see examples of how to set up these tests don't have to discern between special configs that are just for our testing purposes and what is meant for them.

Reply:
> I see! Just to clarify, is the expectation that the same golden file tests will run one way here on our CI and another way when the user creates their own app locally?

Nope! With the current implementation, you can run `flutter create -t list_detail_app my_app` to get an app with some dummy tests, or run `flutter create -t list_detail_app my_app --implementation-tests` to include widget tests that verify the functionality of the app.

My naive idea was that the process could work the exact same way for the CI or end-users.

  > I am wondering, if the same tests are meant to execute differently this way, and we have a special config in the template to run on our CI in flutter/flutter, will that be confusing to users that see that as an example and think it is how they should set up their tests?

Agreed -- that will be super confusing. Given the fact that we need to use Flutter Goldens at this point instead of one consistent process, I'm think it makes sense to only focus on the main goal of testing this app as part of the Flutter CI workflow.

  > Or, would it be possible to create a test in the flutter package that creates this app and keeps a golden file of it (in Gold), and then the template itself can just ship a set of golden file tests that work as the user would expect them?

I can take a deeper look at that option, but my initial gut reaction is that might be a tough sell, since it would require the the `flutter` package to depend on the `flutter_tools` package. What do you think?

May I ask another question: How strongly do folks feel about golden tests for this app? Would it be a bit simpler and just as effective to add more assertions to the suite that verify a new `ThemeMode` is provided to the `MaterialApp` after a user switches Themes from the Settings page?

Reply:
Thanks  for  your help, @Piinks! After playing around with a couple of options, we've decided to go with the simpler route of using plain-old Widget tests for now :)